src/control/server/config/server_test.go

```diff
@@ -129,8 +129,14 @@ func mockConfigFromFile(t *testing.T, path string) (*Server, error) {
 	t.Helper()
 	c := DefaultServer()
 	c.Path = path
+	err := c.Load()
```

Comment:
This is an internal logic specific to PMDK w/ NDCTL. We do not need any parameters, as everything can be derived from the existing configuration. It is a permanent change for PMDK started from version 2.1.0-2 (https://github.com/daos-stack/pmdk/pull/38). We do not want to bother the end-user with something that can be done automatically.

Reply:
The change is fully transparent from an end-user perspective. No changes in configuration, and no changes in behavior except additional NDCTL operations when pmemobj_pool is created/opened/closed. The only thing that we have to adjust are tests as they did not assume such an operation during engine startup like automatic env variables adjustment.
We start with C code modification, moving to a solution by adjusting the default configuration and ending up with the concept of automatic variables update as the algorithm is always the same:
DCPM - bigger default thread stack size (`ABT_THREAD_STACKSIZE`)
RAM - skip ndctl - (`PMEMOBJ_CONF=sds.at_create=0`)
The rest is some sort of validation to avoid configuration that causes server crashes.

Reply:
Thanks for the explanation, whilst the above highlighted change is a workaround that looks out of scope in the test code, it is only test code so I think it's okay.